src/machine.rs

```diff
@@ -0,0 +1,148 @@
+use std::collections::HashMap;
+use std::mem::align_of;
+
+use rustc::mir::{Mir, Local};
+
+use interp::TypedVal;
+
+const X64_ALIGN_SIZE: u8 = 8;
+const X32_ALIGN_SIZE: u8 = 4;
+
+#[derive(Debug)]
+struct Frame<'tcx> {
+    // MIR is a CFG of a single function in rust, so it makes sense
+    // for this to be in the frame.
+    mir: Mir<'tcx>,
+
+    // FIXME: Hacky... I can't find a way to actually access the private u32
+    // inside the `Local` tuple struct. `Local` does  implement the Hashable
+    // and PartialEQ trait however, so for  now it might just work to to use it
+    // as the key
+    locals: HashMap<Local, Vec<u8>>
+}
+
+impl<'tcx> Frame<'tcx> {
+    fn new(mir: Mir<'tcx>) -> Frame {
+        Frame {
+            locals: HashMap::new(),
+            mir:    mir
+        }
+    }
+
+    fn set_local(&mut self, local: Local, val: Vec<u8>) {
+        self.locals.insert(local, val);
+    }
+
+    fn get_local(&self, local: Local) -> &Vec<u8> {
+        self.locals.get(&local).unwrap()
```

Comment:
I think you can just do `self.locals[local]`?

Reply:
Fixed